test/test_torch.py

```diff
@@ -6970,18 +6970,6 @@ def _test_logical(self, device, op, a_, b_, expected_res_):
             for other_dtype in torch.testing.get_all_dtypes():
                 b = torch.tensor(b_, dtype=other_dtype, device=device)
 
-                # Skip bfloat16 on CUDA. Remove this after bfloat16 is supported on CUDA.
```

Comment:
This is still not implemented for CUDA:

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 777, in wrapper
    method(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 777, in wrapper
    method(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 217, in instantiated_test
    return test(self, device_arg)
  File "test_torch.py", line 6998, in test_logical_and
    self._test_logical(device, 'logical_and', [10, 0, 1, 0], [1, 0, 0, 10], [1, 0, 0, 0])
  File "test_torch.py", line 6979, in _test_logical
    self.assertEqual(expected_res.bool(), getattr(a, op)(b))
RuntimeError: "logical_and_cuda" not implemented for 'BFloat16'
```

It's not a type promotion issue. I'd leave this as is.